test/package/install/pvt_pkg_install_test.go

```diff
@@ -0,0 +1,81 @@
+package private_package
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	v1 "github.com/crossplane/crossplane/apis/pkg/v1"
+
+	// corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	pp "github.com/crossplane/test/apis/package"
+)
+
+func TestPackageInstall(t *testing.T) {
+	cases := map[string]struct {
+		reason string
+		body   func(privatePackage string, InstallPackage pp.InstallPackage) error
+	}{
+		"InstallPrivatePackage": {
+			reason: "Should be able to successfully install a private package from registries supported by Crossplane.",
+			body: func(privatePackage string, InstallPackage pp.InstallPackage) error {
+				sl := strings.SplitAfter(privatePackage, "/")
+				InstalledPackageName := sl[len(sl)-1]
+				// sl = strings.SplitAfter(privatePackage, ":")
+				// InstalledPackageVersion := sl
+
+				ctx := context.Background()
+				s := runtime.NewScheme()
+				if err := v1.AddToScheme(s); err != nil {
+					return err
+				}
+
+				c, err := client.New(ctrl.GetConfigOrDie(), client.Options{
+					Scheme: s,
+				})
+				if err != nil {
+					return err
+				}
+				a := resource.NewAPIUpdatingApplicator(c)
+				p := &v1.Configuration{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: InstalledPackageName,
+					},
+					Spec: v1.ConfigurationSpec{
+						PackageSpec: v1.PackageSpec{
+							Package:            InstalledPackageName,
+							PackagePullSecrets: []string{"package-pull-secret"},
+						},
+					},
+				}
+				// Create initial Package
+				if err := a.Apply(ctx, p); err != nil {
+					return err
+				}
+				// Wait for Package to be successfully installed.
+				if err := privatePackage.CheckIfPackageInstalled(ctx, c, 5*time.Second, 2*time.Minute); err != nil {
```

Comment:
We can probably lower the 2 minute timeout here for this test as we should be able to tell fairly quickly whether the package is going to be able to be installed -- 30 seconds would probably give us plenty of cushion 👍🏻

Reply:
noted!